src/main/java/com/mentoring/ecommerce/domain/Product.java

```diff
@@ -0,0 +1,27 @@
+package com.mentoring.ecommerce.domain;
+
+import lombok.Data;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+
+@Entity
+@Data
```

Comment:
@Pelicer, in the dummy project, is used the `@NoArgsConstructor` annotation in the _entity_. The JPA says that the compiler already creates a no-arg constructor. Should I keep it? ([JPA 1.1](https://openjpa.apache.org/builds/1.2.3/apache-openjpa/docs/jpa_overview_pc.html#:~:text=The%20JPA%20specification%20requires%20that,include%20a%20no%2Darg%20constructor.))

The same applies to `response` and `request` DTO's.

Reply:
That's a very interesting point! I didn't know that myself. In that case, only apply a `@NoArgsConstructor` when your class has a constructor already.